assets/js/utils/render-frontend.js

```diff
@@ -2,8 +2,48 @@
  * External dependencies
  */
 import { render } from 'react-dom';
+import { getSetting } from '@woocommerce/settings';
+import { SCHEMA_STORE_KEY } from '@woocommerce/block-data';
+import { useRef } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { forEach } from 'lodash';
 import BlockErrorBoundary from '@woocommerce/base-components/block-error-boundary';
 
+/**
+ * Hydrate Rest API data from settings to reduce the number of API requests needed.
+ *
+ * @param {mixed} children Child components.
+ */
+const HydrateRestApiData = ( { children } ) => {
+	const restApiRoutes = useRef( getSetting( 'restApiRoutes' ) );
+
+	useSelect( ( select, registry ) => {
+		const { isResolving, hasFinishedResolution } = select(
+			SCHEMA_STORE_KEY
+		);
+		const {
+			receiveRoutes,
+			startResolution,
+			finishResolution,
+		} = registry.dispatch( SCHEMA_STORE_KEY );
+		if ( ! restApiRoutes.current ) {
+			return;
+		}
```

Comment:
My bad, but we could probably move this condition to the top of the function block so it bails early?